### DISCUSSION TITLE
Question about Chassis Device Type

### DISCUSSION THREAD

**Discussion by nathaniel-steele** [opendcim/openDCIM#1513] [Q&A]
Hello, new user here. I have looked over all documentation I can find and did not see any info on how the chassis device type works.

I have a Number of appliance devices setting on a rack shelf, they occupy 3U of vertical space and I can fit 10 across the shelf. I created a Chassis device called Shelf and gave it a height of 3 and 10 front slots. I have added devices to some slots. I do not get images of the devices installed in the slots. Is there something I am doing incorrectly. will upload screenshots momentarily.

**Comment by ArumBlack** [comment]
<img width="379" alt="image" src="https://github.com/opendcim/openDCIM/assets/120606523/07f81f7a-d9ce-4579-b00a-961ff1910df3">

<img width="376" alt="image" src="https://github.com/opendcim/openDCIM/assets/120606523/b8b85da4-52ac-41df-9fcf-cd48f5879835">

<img width="384" alt="image" src="https://github.com/opendcim/openDCIM/assets/120606523/2abf81cb-732f-4269-8c4a-8c31ee88b50b">

**Reply by ArumBlack** [reply]
I have made some assumptions about how I think it should work, anyone see what I am doing wrong?

**Reply by ArumBlack** [reply]
I also tried with no image for the chassis, and with the device images rotated 90 degrees (orientation shown is correct and slots should go left to right. I don't see anything to indicate which way slots go, or anyway to have a number of rows of slots either, which is fine for me unless we are limited to vertically stacked slots only, which I guess I could make work, but the images would not be a photorealistic representation then.

**Reply by ArumBlack** [reply]
And apparently, I have 2 different Github accounts. I posted the original question from a different computer.

**Comment by wilpig** [comment]
Edit the template for the chassis device and use the slot editor to define your slots on the picture of your chassis. Then your individual devices pictures will show in the slots you define

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by ArumBlack** [reply]
Perfect thank you. Somehow my brain completely ignored the button marked edit coordinates on the template editor....Picture below for future reference if anybody ever has the same issue and searches and finds this thread.

<img width="578" alt="image" src="https://github.com/opendcim/openDCIM/assets/120606523/dbfa31a8-70b5-44cc-8169-f3c47377a91e">

**Reply by ArumBlack** [reply]
Is it possible to populate Chassis Slots via a bulk upload? I've got 25 Chassis, with 8 Devices in each one. Most of the data already exists in a spreadsheet I can copy/paste. the rest would be easier to enter into the bulk import spreadsheet, rather than 1 device at a time in the GUI.

**Comment by alex001x** [comment]
No, drawing the frame provides a click link area to load the child device, you can import the devices in bulk and or the library but you must manually associate the images. the association of child devices in the chassis also do manually. 
Sorry my english is not good ! 
Alexandre 
FR

**Reply by ArumBlack** [reply]
well....over a year later and I am finally trying this. I was able to import devices but do not understand how to associate them with the chassis. currently they are added as zero u devices as I specified a rack but not a space.
![image](https://github.com/user-attachments/assets/bf540332-a91d-4e5e-b403-b5ff201590a3)

When I go to the chassis and click Add device the only option I have is to create one.

![image](https://github.com/user-attachments/assets/dcd4ea55-4407-4e01-842d-9f026a200f87)
![image](https://github.com/user-attachments/assets/b4514898-3593-4852-91b4-6cef0e850812)


It would be nice if there was an option to add an existing device.

It would be even nicer if they could bulk import, all that would be needed is to add a field to the bulk import to specify the parent device.

But I am not a programmer so maybe it is far more complicated than that!

I have no desire to go messing in the database directly.

**Reply by ArumBlack** [reply]
Sam answered this here:
https://github.com/opendcim/openDCIM/discussions/1605

enter the position as ParentU.Slot, so in my example above the next Module is 31.3.

**Comment by ArumBlack** [comment]
answered
